src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -174,73 +175,86 @@ export const useWorkspace = (namespace, name): WorkspaceDetails => {
-      }
+    // Request a service account token. If this is the first time, it could take some time before everything is in sync.
+    // Doing this now, even though we don't explicitly need it now, increases the likelihood that it will be ready when it is needed.
+    if (canWrite(accessLevel) && isGoogleWorkspace(workspace)) {
+      saToken(googleProject);
+    }
 
-      // This is old code-- it is unclear if this case can actually happen anymore.
-      if (!isOwner(accessLevel) && createdBy === getTerraUser().email && differenceFromNowInSeconds(createdDate) < 60) {
-        accessNotificationId.current = notify('info', 'Workspace access synchronizing', {
-          message: h(Fragment, [
-            'It looks like you just created this workspace. It may take up to a minute before you have access to modify it. Refresh at any time to re-check.',
-            div({ style: { marginTop: '1rem' } }, [
-              h(
-                Link,
-                {
-                  onClick: () => {
-                    refreshWorkspace();
-                    clearNotification(accessNotificationId.current);
-                  },
+    // This is old code-- it is unclear if this case can actually happen anymore.
+    if (!isOwner(accessLevel) && createdBy === getTerraUser().email && differenceFromNowInSeconds(createdDate) < 60) {
+      accessNotificationId.current = notify('info', 'Workspace access synchronizing', {
+        message: h(Fragment, [
+          'It looks like you just created this workspace. It may take up to a minute before you have access to modify it. Refresh at any time to re-check.',
+          div({ style: { marginTop: '1rem' } }, [
+            h(
+              Link,
+              {
+                onClick: () => {
+                  refreshWorkspace();
+                  clearNotification(accessNotificationId.current);
                 },
-                ['Click to refresh now']
-              ),
-            ]),
+              },
+              ['Click to refresh now']
+            ),
           ]),
-        });
-      }
-    } catch (error: any) {
-      if (error.status === 404) {
-        setAccessError(true);
-      } else {
-        throw error;
-      }
+        ]),
+      });
     }
-  }) as () => Promise<void>;
+  };
+
+  const checkForAccessError: ErrorCallback = (error: unknown) => {
+    if (error instanceof Response && error.status === 404) {
+      setAccessError(true);
+    } else {
+      throw error;
+    }
+  };
+
+  const refreshWorkspace: () => Promise<void> = _.flow(
+    withErrorHandling(checkForAccessError),
+    withErrorReporting('Error loading workspace'),
+    withBusyState(setLoadingWorkspace)
+  )(doWorkspaceRefresh) as () => Promise<void>;
+
+  // refresh the workspace without triggering busy indicators
+  // if an error handling function is passed, use that - otherwise ignpre errors
+  const silentlyRefreshWorkspace = (errorHandling?: ErrorCallback): Promise<void> => {
```

Comment:
The default is to ignore errors, since this is intended to be a background process, and the workspace has already loaded successfully the first time.

The custom callback is so that the WorkspaceContainer can redirect to the list view when the workspace is deleted.